pkg/virt-controller/watch/vm.go

```diff
@@ -3457,15 +3457,15 @@ func setGuestMemory(spec *virtv1.VirtualMachineInstanceSpec) {
 		return
 	}
 
+	if spec.Domain.Memory == nil {
+		spec.Domain.Memory = &virtv1.Memory{}
+	}
+
 	switch {
 	case !spec.Domain.Resources.Requests.Memory().IsZero():
-		spec.Domain.Memory = &virtv1.Memory{
-			Guest: spec.Domain.Resources.Requests.Memory(),
-		}
+		spec.Domain.Memory.Guest = spec.Domain.Resources.Requests.Memory()
 	case !spec.Domain.Resources.Limits.Memory().IsZero():
-		spec.Domain.Memory = &virtv1.Memory{
-			Guest: spec.Domain.Resources.Limits.Memory(),
-		}
+		spec.Domain.Memory.Guest = spec.Domain.Resources.Limits.Memory()
 	case spec.Domain.Memory != nil && spec.Domain.Memory.Hugepages != nil:
```

Comment:
I think we can drop `spec.Domain.Memory != nil` now

Reply:
ah yes, you're right.